tests/test_examples.py

```diff
@@ -361,3 +214,141 @@ def test_water_sampling_mc_buckyball(batch_size, insertion_type):
+            proc = run_example("run_rbfe_legs.py", get_cli_args(config_a), cwd=temp_dir)
+            assert proc.returncode == 0
+            verify_run(Path(temp_dir) / config_a["output_dir"])
+
+            config_b = dict(output_dir="b", **config)
+            assert config_b["output_dir"] != config_a["output_dir"], "Runs are writing to the same output directory"
+            proc = run_example("run_rbfe_legs.py", get_cli_args(config_b), cwd=temp_dir)
+            assert proc.returncode == 0
+            verify_run(Path(temp_dir) / config_b["output_dir"])
+
+            def verify_simulations_match(ref_dir: Path, comp_dir: Path):
+                with open(ref_dir / "md_params.pkl", "rb") as ifs:
+                    ref_md_params = pickle.load(ifs)
+                with open(comp_dir / "md_params.pkl", "rb") as ifs:
+                    comp_md_params = pickle.load(ifs)
+                assert ref_md_params == comp_md_params, "MD Parameters don't match"
+
+                with open(ref_dir / "core.pkl", "rb") as ifs:
+                    ref_core = pickle.load(ifs)
+                with open(comp_dir / "core.pkl", "rb") as ifs:
+                    comp_core = pickle.load(ifs)
+                assert np.all(ref_core == comp_core), "Atom mappings don't match"
+
+                ref_results = np.load(str(ref_dir / leg / "results.npz"))
+                comp_results = np.load(str(comp_dir / leg / "results.npz"))
+                np.testing.assert_equal(ref_results["pred_dg"], comp_results["pred_dg"])
+                np.testing.assert_equal(ref_results["pred_dg_err"], comp_results["pred_dg_err"])
+                np.testing.assert_array_equal(ref_results["overlaps"], comp_results["overlaps"])
+                np.testing.assert_equal(ref_results["n_windows"], comp_results["n_windows"])
+
+                with open(ref_dir / leg / "simulation_result.pkl", "rb") as ifs:
+                    ref_res = pickle.load(ifs)
+                with open(comp_dir / leg / "simulation_result.pkl", "rb") as ifs:
+                    comp_res = pickle.load(ifs)
+                assert len(ref_res.final_result.initial_states) == ref_results["n_windows"]
+                assert len(ref_res.final_result.initial_states) == len(comp_res.final_result.initial_states)
+
+                for ref_state, comp_state in zip(
+                    ref_res.final_result.initial_states, comp_res.final_result.initial_states
+                ):
+                    np.testing.assert_array_equal(ref_state.x0, comp_state.x0)
+                    np.testing.assert_array_equal(ref_state.v0, comp_state.v0)
+                    np.testing.assert_array_equal(ref_state.box0, comp_state.box0)
+                    np.testing.assert_array_equal(ref_state.ligand_idxs, comp_state.ligand_idxs)
+                    np.testing.assert_array_equal(ref_state.protein_idxs, comp_state.protein_idxs)
+                    assert len(ref_state.potentials) == len(comp_state.potentials)
```

Comment:
nit: might be overkill, but could you use [`assert_deep_eq`](https://github.com/proteneer/timemachine/blob/63e601360c1c05005c801690d197d6b922f46284/timemachine/fe/free_energy.py#L858) here?